proxy.go

```diff
@@ -38,8 +41,8 @@ var errProxyTimeout = errors.New("all peers timed out")
 //     cancels any running requests.
 func (vs *version) proxy(r *http.Request, peers []string) (*http.Response, string, error) {
 	responses := make(chan proxyResponse, len(peers))
-	ctx, _ := context.WithDeadline(r.Context(),
-		time.Now().Add(vs.sequins.config.Sharding.ProxyTimeout.Duration))
+	totalTimeout := time.NewTimer(vs.sequins.config.Sharding.ProxyTimeout.Duration)
+	ctx, cancel := context.WithCancel(r.Context())
```

Comment:
I think you want to `defer cancel()`—otherwise I'm pretty sure you leak resources